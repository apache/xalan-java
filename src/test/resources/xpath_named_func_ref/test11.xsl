<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
                xmlns:xs="http://www.w3.org/2001/XMLSchema"
                exclude-result-prefixes="xs"				
                version="3.0">
                
    <!-- Author: mukulg@apache.org -->
    
    <!-- An XSL stylesheet test case, to test XML Schema constructor functions 
         for schema built-in types used in two different ways. One as XPath named 
         function reference and called via XPath dynamic function call syntax, 
         and second with usual XML Schema constructor function call syntax. -->                
 
    <xsl:output method="xml" indent="yes"/>
 
    <xsl:template name="main">
       <result>
          <one>
			  <xsl:variable name="dateFunc" select="xs:date#1" as="function(*)"/>
			  <a>	                
				 <xsl:value-of select="current-date() gt $dateFunc('2005-10-05')"/>
			  </a>
			  <b>	                
				 <xsl:value-of select="current-date() lt $dateFunc('2005-10-05')"/>
			  </b>
			  <c>
				 <xsl:value-of select="$dateFunc('2005-10-05') + xs:dayTimeDuration('P2DT2H30M0S')"/>
			  </c>
		  </one>	   
		  <two>
			  <a>	                
				 <xsl:value-of select="current-date() gt xs:date('2005-10-05')"/>
			  </a>
			  <b>	                
				 <xsl:value-of select="current-date() lt xs:date('2005-10-05')"/>
			  </b>
			  <c>
				 <xsl:value-of select="xs:date('2005-10-05') + xs:dayTimeDuration('P2DT2H30M0S')"/>
			  </c>
		  </two>		  
       </result>
    </xsl:template>
    
    <!--
      * Licensed to the Apache Software Foundation (ASF) under one
      * or more contributor license agreements. See the NOTICE file
      * distributed with this work for additional information
      * regarding copyright ownership. The ASF licenses this file
      * to you under the Apache License, Version 2.0 (the "License");
      * you may not use this file except in compliance with the License.
      * You may obtain a copy of the License at
      *
      *     http://www.apache.org/licenses/LICENSE-2.0
      *
      * Unless required by applicable law or agreed to in writing, software
      * distributed under the License is distributed on an "AS IS" BASIS,
      * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
      * See the License for the specific language governing permissions and
      * limitations under the License.
    -->
	
</xsl:stylesheet>
