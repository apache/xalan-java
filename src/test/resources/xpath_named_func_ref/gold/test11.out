<?xml version="1.0" encoding="UTF-8"?><result>
  <one>
    <a>true</a>
    <b>false</b>
    <c>2005-10-07</c>
  </one>
  <two>
    <a>true</a>
    <b>false</b>
    <c>2005-10-07</c>
  </two>
</result>
